Dont look at reduction output args when computing mem dependencies

bertmaher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #52196 Remove ReduceOp::accumulator
* #52187 [nnc] Remove output_args from ReduceOp
* #52177 Avoid using ReduceOp->output_args() in rfactor
* **#52170 Dont look at reduction output args when computing mem dependencies**

We don't actually need to pull the Reduce output args here, because they're just coming from the Load/Store indices anyways.

Differential Revision: [D26411735](https://our.internmc.facebook.com/intern/diff/D26411735/)

bertmaher committed:
Dont look at reduction output args when computing mem dependencies



Differential Revision: [D26411735](https://our.internmc.facebook.com/intern/diff/D26411735/)

[ghstack-poisoned]

bertmaher committed:
Update on "Dont look at reduction output args when computing mem dependencies"


We don't actually need to pull the Reduce output args here, because they're just coming from the Load/Store indices anyways.

Differential Revision: [D26411735](https://our.internmc.facebook.com/intern/diff/D26411735/)

[ghstack-poisoned]

navahgar commented on torch/csrc/jit/tensorexpr/mem_dependency_checker.cpp:
```diff
@@ -64,8 +64,6 @@ std::vector<const Expr*> AccessInfo::getIndices() const {
       indices = load->indices();
     } else if (auto* call = dynamic_cast<const FunctionCall*>(expr_)) {
       indices = call->params();
-    } else if (auto* reduce = dynamic_cast<const ReduceOp*>(expr_)) {
```
This is assuming that the dependency analysis does not need to happen at the Tensor level. Is that always guaranteed now? I understand that, if we do the cleanup as @ZolotukhinM suggested, then we are going to end up there. But if we use this dependency analysis before lowering to stmt now, I think we will need this. Am I missing something?

navahgar replied:
It looks like loads and stores are present even before lowering ReduceOp completely. So, this should be fine AFAICT.

navahgar reviewed (approved):
LGTM

bertmaher committed:
fix clang-tidy on "Dont look at reduction output args when computing mem dependencies"


We don't actually need to pull the Reduce output args here, because they're just coming from the Load/Store indices anyways.

Differential Revision: [D26411735](https://our.internmc.facebook.com/intern/diff/D26411735/)

[ghstack-poisoned]
